src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs

```diff
@@ -752,6 +752,34 @@ private static InstructionSetFlags ExpandInstructionSetByReverseImplicationHelpe
             return resultflags;
         }
 
+        private static Dictionary<(string, TargetArchitecture), string> AllInstructionSetGroups { get; } = new()
+            {
+                { ("x86-x64",    TargetArchitecture.X64),   "sse2" },
+                { ("x86-x64",    TargetArchitecture.X86),   "sse2" },
+                { ("x86-x64-v2", TargetArchitecture.X64),   "sse4.2 popcnt aes" },
+                { ("x86-x64-v2", TargetArchitecture.X86),   "sse4.2 popcnt aes" },
+                { ("x86-x64-v3", TargetArchitecture.X64),   "x86-x64-v2 avx2 bmi2 lznct pclmul movbe fma" },
+                { ("x86-x64-v3", TargetArchitecture.X86),   "x86-x64-v2 avx2 bmi2 lznct pclmul movbe fma" },
+                { ("skylake",    TargetArchitecture.X64),   "x86-x64-v2 avx2 bmi2 lznct pclmul movbe fma" },
+                { ("skylake",    TargetArchitecture.X86),   "x86-x64-v2 avx2 bmi2 lznct pclmul movbe fma" },
+                { ("x86-x64-v4", TargetArchitecture.X64),   "x86-x64-v3 avxvnni serialize" },
```

Comment:
`avxvnni` and `serialize` aren't implied by `v4`

`serialize` is a brand new instruction that is, I believe, only available on Alder Lake and newer.

`AVX-VNNI` is likewise "newer", it is not part of AVX-512 but rather a "backport" of the AVX-VNNI functionality to the VEX encoding (where as AVX512-VNNI is distinct and uses the EVEX encoding).